### PR TITLE
Improve troubleshooting docs

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**
Improve troubleshooting docs to include mitigation steps, investigation steps, and basic debugging tips for InstrumentationMethod owners.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
+
+When an issue occurs in production, the first step is to mitigate it so the application continues to run. A common cause of problems are from updates to either CLRIE or to one of the Instrumentation Method client that introduces potential conflicts with another client or incompatiblities with the app.
+
+### Remove CLRIE
+
+The most direct way to stop CLRIE and all Instrumentation Methods is to remove the ICorProfiler environment variables that are being applied to the process (along with a process restart). This may either require updating applicationHost.xdt files, changing the `InstrumentationEngine_EXTENSION_VERSION` Application Setting, or modifying registry keys. Without the ICorProfiler environment variables, the process effectively removes CLRIE & all IMs that are currently running. If the application without CLRIE is still encountering errors, then the issue is something the app owner will need to address.
+
+Alternatively, if the user cannot modify the ICorProfiler environment variable, they may be able to at least remove the dlls which are pointed to by the COR_PROFILER_PATH/CORECLR_PROFILER_PATH environment variables. A process restart will still be required.
```
Comment (wiktork) [inline docs/troubleshooting.md:15]: Consider going over the WAD specific mitigation; it's challenging to alter the environment block manually.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -25,11 +53,25 @@ If `MicrosoftInstrumentationEngine_FileLogPath` is set along with `MicrosoftInst
...
+
+#### Breakpoint Locations
+
+The best place to start debugging is in [ProfilerManager.cpp](../src/InstrumentationEngine/ProfilerManager.cpp). ProfilerManager is CLRIE's ICorProfiler implementation which gets the first-chance callback during profiling. It controls setting up InstrumentationMethods and forwarding ICorProfiler callback events to them.
+
+Setting a breakpoint inside `CProfilerManager::Initialize()` provides the earliest point where CLRIE is beginning to initialize in the process.
+
+`CProfilerManager::AddInstrumentationMethod()` provides the logic for loading each InstrumentationMethod and calling Initialize() on them.
+
+CLRIE's main purpose is to coordinate instrumentation, so callbacks such as `CProfilerManager::JITCompilationStarted()` and `CProfilerManager::GetReJITParameters()` are also good places to inspect.
```
Comment (wiktork) [inline docs/troubleshooting.md:77]: I'm wondering if we can develop a CLR debugging workflow based on public symbol servers for CLR? This way, users can try to debug their il failures at the point of failure (e.g. ProgramInvalidException).

Comment (WilliamXieMSFT) [reply docs/troubleshooting.md:77]: I can at least add some details about public symbol servers.

**Pull request merged**